packages/dev/sharedUiComponents/src/fluent/primitives/colorPicker.tsx

```diff
@@ -0,0 +1,407 @@
+/* eslint-disable jsdoc/require-returns */
+/* eslint-disable @typescript-eslint/naming-convention */
+import * as React from "react";
+import { ColorPicker, ColorSlider, ColorArea, AlphaSlider, InfoLabel, Link, makeStyles, Popover, PopoverSurface, PopoverTrigger, tokens } from "@fluentui/react-components";
+import type { ColorPickerProps as FluentColorPickerProps } from "@fluentui/react-components";
+
+import { Input, type InputProps, Label, SpinButton, type SpinButtonChangeEvent, type SpinButtonOnChangeData, type SpinButtonProps, useId } from "@fluentui/react-components";
+import { Color3, Color4 } from "core/Maths";
+import type { BaseComponentProps } from "../hoc/propertyLine";
+
+export type ColorPickerProps<C extends Color3 | Color4> = {
+    linearHint?: boolean;
+} & BaseComponentProps<C>;
+
+const useColorPickerStyles = makeStyles({
+    colorPickerContainer: {
+        width: "300px",
+        display: "flex",
+        flexDirection: "column",
+        gap: "10px",
+        overflow: "visible",
+    },
+    triggerColor: {
+        width: "50px",
+        height: tokens.lineHeightBase100,
+        borderRadius: tokens.borderRadiusSmall,
+        border: "1px solid #ccc",
+        "@media (forced-colors: active)": {
+            forcedColorAdjust: "none",
+        },
+    },
+    previewColor: {
+        width: "50px",
+        height: "50px",
+        borderRadius: tokens.borderRadiusSmall,
+        border: "1px solid #ccc",
+        "@media (forced-colors: active)": {
+            forcedColorAdjust: "none",
+        },
+    },
+    inputFields: {
+        display: "flex",
+        alignItems: "flex-end",
+        flexDirection: "row",
+        gap: "10px",
+    },
+    colorFieldWrapper: {
+        display: "flex",
+        flexDirection: "column",
+        gap: "6px",
+    },
+    input: {
+        width: "80px",
+    },
+    spinButton: {
+        minWidth: "60px",
+    },
+    container: {
+        display: "flex",
+        flexDirection: "column",
+        gap: "16px",
+    },
+    row: {
+        display: "flex",
+        flexDirection: "row",
+        gap: "12px",
+        alignItems: "center",
+    },
+});
+
+export const ColorPickerPopup = (props: ColorPickerProps<Color3 | Color4>) => {

```

Comment:
I would be explicit and declare all these components as `FunctionComponent<...>` for consistency and error checking.